hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/NamenodeHeartbeatService.java

```diff
@@ -170,7 +172,20 @@ protected void serviceInit(Configuration configuration) throws Exception {
 
   @Override
   public void periodicInvoke() {
-    updateState();
+    try {
+      SecurityUtil.doAsCurrentUser(
+          new PrivilegedExceptionAction<Object>() {
```

Comment:
Can this be a lambda?

Reply:
Sure, agreed that'll be more readable here.

Reply:
Thanks @goiri and @bolerio for your comments. I am just concern if is it necessary to do as current user here because Router has login when start daemon and it is already execute with current login user. Do you meet some exception here? Thanks.